### ISSUE TITLE
Hue Motion Sensors update stuck at queued

### ISSUE THREAD

**Issue by woytekbode** [dresden-elektronik/deconz-ota-plugin#27]
I recently installed Home Assistant on a Raspberry Pi4 with SSD and conbee 2. Running the Deconz add-on and want to have hue lights, motion sensors and aqara sensors on the network.

When I found out it is possible to update devices over the air with the Deconz addon it made me very happy. That means it's not necessary to keep an old Hue Hub present for updates. However, I'm having some problems with updating my Hue motion sensors.

Steps that I took:
- Download Hue motion sensor firmware 6.1.1.27575 from the link in the readme
- Place the firmware in the HA share folder with WinSCP
- Copy the firmware to the docker data/otau folder with SSH
- Start the STD OTAU plugin from Phoscon
- Select the correct device
- Select the firmware
- Select Update

The update is recognised and seems to start but it hangs on "queued" and the time stops at 0.46. Things I have tried:
- Let Phoscon rename the file to 100B-010D-42006BB7.zigbee and try again
- Restart the motion sensor by pulling and reinserting the battery
- Wait overnight
- Retry pressing the query/update buttons...

![Otau_update_queued](https://user-images.githubusercontent.com/67194886/117553851-25a14700-b054-11eb-8e36-913e691e13eb.PNG)

**Comment by woytekbode** [comment]
So I have managed to update my Hue motion sensors. I did the following:

- Disconnect the Hue light so that the sensors are forced to talk to the coordinator directly (note that I only have a test set-up right now with one light and several sensors, doing this in an actual house set-up will be much more difficult)
- Manually select the firmware that was renamed by restarting Deconz
- Press the upload button
- Press the query button

I'm glad I got it to work but I am wondering how I can do the update when the sensors are located all over the house with routers everywhere in between. It's not so easy to force a direct connection with the coordinator in that situation.

Maybe we can try and figure out if something is wrong when updating via routers?

**Comment by jonnycastaway** [comment]
Same issue with all Hue Devices.
Hue Devices, no matter if it's a Bulb, Dimmer, Motion Sensor, stuck at queued or at 0.03%.
Ikea Devices updates without problems.
Biggest Problem ist that stuck at 0.03% will block the complete updates because deconz handles updates one by one.
Bring the device near to the coordinator is not an option because i have bulbs that i cannot move.

**Comment by matlimatli** [comment]
I agree that it seems to be an issue with updating via routers. I managed to shut off various lights around the house and thereby forcing my dimmer switch to connect directly to the coordinator. The update then went smoothly, until after ~90%, when the device suddenly connected to a different router. After this, the update didn't make any progress at all.

**Comment by NePheus** [comment]
Same problem here. Is this a fixable bug by the dev team? Or cant this behavior be changed and we have to force (somehow) the connection directly to the coordinator?

**Comment by ebaauw** [comment]
Do you guys have source routing enabled?  If so, could you disable it, wait a couple of minutes and try again?

**Comment by jdeluyck** [comment]
> Do you guys have source routing enabled? If so, could you disable it, wait a couple of minutes and try again?

Where can I find this? I've been peeking through deconz, but...

**Comment by ebaauw** [comment]
In the GUI, enable the _Source Routing_ panel under _Panels_.

**Comment by jdeluyck** [comment]
Not enabled here. (and Hue doesn't update unless it's directly linked)

**Comment by ebaauw** [comment]
What coordinator device and firmware do you use?

**Comment by jdeluyck** [comment]
ConbeeII running firmware 26660700, home assistant addon.

**Comment by antonbo** [comment]
I was able to proceed a little by using instructions from #22. But it still stuck for me at 4-5%. And restarted from 0% if I press Query or Update

**Comment by antonbo** [comment]
So all my sensors are updated now. The procedure involved 'xdotool click 1' to force Touchlink scan every 60-90 seconds via old UI settings. Be careful to avoid OTA plugin generated images. In my case they were corrupted so I've used original ones from Philips site. I've tried to do the same with dimmer switches, but it fails with unknown reason for me. The firmware was sent fully to the device, but after restart (and few red-green blinks on device) it still shows the old one.

It's also possible to request Touchlink scan via REST API, but I have not tried it.

**Comment by woytekbode** [comment]
@ebaauw Hi Erik. IS there something we can do to progress this? I'd be happy to run some tests to find out where the issue is. I can imagine it's not high priority but on the long run it would be very inconvenient if firmware updates are not possible when using Deconz instead of the Hue Hub.

@antonbo Nice to see you have made some progress!

**Comment by smallgreyrock** [comment]
Disabling source routing made the OTAU feature mostly work on my 50-60 device network, albeit not very reliably.

One frustrating thing about Hue devices is that if they're sufficiently out of date you seem to have to update to a different old firmware (or multiple different old firmwares) before you can update to the current one. I don't know if deconz handles that gracefully, you may have to select a specific firmware file in that case.

Sometimes the update check fails for reasons I don't understand -- I've seen deconz reply OTAU_NO_IMAGE_AVAILABLE for querying devices that definitely do have an updated image available.

Trying to force the device to start updating using query/update is does not seem to work reliably.

Leaving deconz alone overnight updated ~90% of the devices. The remaining 10% were either stuck in queued or at some random percentage. Clicking abort/query/update a couple of times sometimes unstuck them.

I wish disabling source routing was more clearly documented, I'd never have guessed that was breaking OTAU if I hadn't found this ticket. For anyone else who stumbles on this, OTAU does work, you just ... might have to wait and flail around a bit.
